src/main/java/chess/service/ChessGameService.java

```diff
@@ -1,42 +1,77 @@
 package chess.service;
 
 import chess.dao.ChessGameDao;
-import chess.dao.ChessGameDaoImpl;
+import chess.dao.PieceDao;
 import chess.dao.dto.ChessGameDto;
+import chess.dao.dto.PieceDto;
+import chess.domain.board.Board;
+import chess.domain.board.Square;
 import chess.domain.game.ChessGame;
+import chess.domain.piece.Piece;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
 public class ChessGameService {
 
-    private static final ChessGameDao chessGameDao = new ChessGameDaoImpl();
+    private final ChessGameDao chessGameDao;
+    private final PieceDao pieceDao;
 
-    public void save(final ChessGame chessGame) {
-        chessGameDao.save(chessGame);
+    public ChessGameService(final ChessGameDao chessGameDao, final PieceDao pieceDao) {
+        this.chessGameDao = chessGameDao;
+        this.pieceDao = pieceDao;
     }
 
-    public List<Long> findAllChessGameId() {
+    public List<Long> loadAllChessGameId() {
         final List<ChessGameDto> chessGames = chessGameDao.findAll();
         return chessGames.stream()
                 .map(ChessGameDto::getId)
                 .collect(Collectors.toList());
     }
 
-    public ChessGameDto findById(final Long id) {
-        return chessGameDao.findById(id)
-                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 게임아이디 입니다."));
+    public ChessGame loadChessGameById(final Long chessGameId) {
+        final ChessGameDto chessGame = chessGameDao.findById(chessGameId)
+                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 게임 ID 입니다."));
+        final List<PieceDto> pieces = pieceDao.findAllByChessGameId(chessGameId);
+        return ChessGameFactory.create(chessGame, pieces);
+    }
+
+    public void saveChessGame(final ChessGame chessGame) {
+        chessGameDao.save(chessGame);
+        final ChessGameDto chessGameDto = chessGameDao.findLatest()
+                .orElseThrow(() -> new IllegalStateException("저장된 게임이 존재해야 합니다."));
+        chessGame.setId(chessGameDto.getId());
+        savePieces(chessGame);
```

Comment:
```suggestion
    public void saveChessGame(final ChessGame chessGame) {
        chessGameId = chessGameDao.save(chessGame);
        final Map<Square, Piece> table = chessGame.getBoard();
        for (final Square square : table.keySet()) {
            pieceDao.save(chessGameId, square, table.get(square));
        }
```

id 값을 구하기 위해 복잡해진 것 같네요! 아래 방식을 통해 dao level에서 id 를 반환하거나, id 값이 추가된 객체를 반환해볼 수도 있을 것 같아요! (db 로직을 위해 서비스가 복잡해진 것으로 보입니다)

https://www.springcloud.io/post/2022-06/jdbctemplate-id/#gsc.tab=0